packages/tokenbank_v1/contracts/modules/loopring/ExchangeModule.sol

```diff
@@ -0,0 +1,247 @@
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.5.11;
+
+import "../../iface/Wallet.sol";
+
+import "../stores/ReimbursementStore.sol";
+
+import "../security/SecurityModule.sol";
+
+import "./IExchangeV3.sol";
+
+
+/// @title ExchangeModule
+contract ExchangeModule is SecurityModule
+{
+    event AccountUpdated(
+        address indexed exchange,
+        address indexed wallet,
+        bool            creation
+    );
+
+    event Deposit(
+        address indexed exchange,
+        address indexed wallet,
+        address indexed token,
+        uint96          amount
+    );
+
+    event Withdrawal(
+        address indexed exchange,
+        address indexed wallet,
+        address indexed token,
+        uint96          amount
+    );
+
+    ReimbursementStore internal reimbursementStore;
+
+    constructor(
+        SecurityStore      _securityStore,
+        ReimbursementStore _reimbursementStore
+        )
+        public
+        SecurityModule(_securityStore)
+    {
+        reimbursementStore = _reimbursementStore;
+    }
+
+    function staticMethods()
+        public
+        pure
+        returns (bytes4[] memory methods)
+    {
+        methods = new bytes4[](3);
+        methods[0] = this.isExchangeRunning.selector;
+        methods[1] = this.getDEXSlots.selector;
+        methods[2] = this.getDEXAccount.selector;
+    }
+
+    function isExchangeRunning(IExchangeV3 exchange)
+        public
+        view
+        returns (bool)
+    {
+        return (
+            !exchange.isInMaintenance() &&
+            !exchange.isInWithdrawalMode() &&
+            !exchange.isShutdown()
+        );
+    }
+
+    function getDEXSlots(IExchangeV3 exchange)
+        public
+        view
+        returns (
+            uint numAvailableDepositSlots,
+            uint numAvailableWithdrawalSlots
+        )
+    {
+        (,numAvailableDepositSlots, , numAvailableWithdrawalSlots) = exchange.getRequestStats();
+    }
+
+    function getDEXAccount(IExchangeV3 exchange)
+        public
+        view
+        returns (
+            uint24 accountId,
+            uint   pubKeyX,
+            uint   pubKeyY
+        )
+    {
+        return exchange.getAccount(address(this));
+    }
+
+    function createOrUpdateDEXAccount(
+        address payable wallet,
+        IExchangeV3     exchange,
+        uint            pubKeyX,
+        uint            pubKeyY,
+        bytes calldata  permission,
+        address         feeToken,
+        uint            feeAmount
+        )
+        external
+        nonReentrantExceptFromThis
+        onlyFromMetaTxOrWalletOwner(wallet)
+        onlyWhenWalletUnlocked(wallet)
+    {
+        (uint creationFee, uint updateFee, , ) = exchange.getFees();
+        (uint24 accountId, , ) = exchange.getAccount(wallet);
```

Comment:
This will actual throw when the account doesn't exist. Maybe we should change that so it's easier to use from smart contracts.